From a1e99ffe54b823f839a0a7be4fcb54169aa3241c Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-a1e99ffe54b-20221108T134102Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sat, 26 Mar 2022 14:01:06 +0100
Subject: [PATCH] revisions API: extend the nascent REV_INFO_INIT macro
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Have the REV_INFO_INIT macro added in f196c1e908d (revisions API
users: use release_revisions() needing REV_INFO_INIT, 2022-04-13)
initialize more fields, rather than having repo_init_revisions() do
so.

This isn't sufficient to initialize a "struct rev_info" yet, but we
are getting closer to that eventual goal. Even though we can't fully
initialize a "struct rev_info" with REV_INFO_INIT it's useful to
clearly separate those things that we can statically initialize, and
those that we can't.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 revision.c | 20 ++------------------
 revision.h | 19 ++++++++++++++++++-
 2 files changed, 20 insertions(+), 19 deletions(-)

diff --git a/revision.c b/revision.c
index 0760e78936e..84094b96332 100644
--- a/revision.c
+++ b/revision.c
@@ -1865,30 +1865,15 @@ void repo_init_revisions(struct repository *r,
 			 struct rev_info *revs,
 			 const char *prefix)
 {
-	memset(revs, 0, sizeof(*revs));
+	struct rev_info blank = REV_INFO_INIT;
+	memcpy(revs, &blank, sizeof(*revs));
 
 	revs->repo = r;
-	revs->abbrev = DEFAULT_ABBREV;
-	revs->simplify_history = 1;
 	revs->pruning.repo = r;
-	revs->pruning.flags.recursive = 1;
-	revs->pruning.flags.quick = 1;
 	revs->pruning.add_remove = file_add_remove;
 	revs->pruning.change = file_change;
 	revs->pruning.change_fn_data = revs;
-	revs->sort_order = REV_SORT_IN_GRAPH_ORDER;
-	revs->dense = 1;
 	revs->prefix = prefix;
-	revs->max_age = -1;
-	revs->max_age_as_filter = -1;
-	revs->min_age = -1;
-	revs->skip_count = -1;
-	revs->max_count = -1;
-	revs->max_parents = -1;
-	revs->expand_tabs_in_log = -1;
-
-	revs->commit_format = CMIT_FMT_DEFAULT;
-	revs->expand_tabs_in_log_default = 8;
 
 	grep_init(&revs->grep_filter, revs->repo);
 	revs->grep_filter.status_only = 1;
@@ -1900,7 +1885,6 @@ void repo_init_revisions(struct repository *r,
 	}
 
 	init_display_notes(&revs->notes_opt);
-	list_objects_filter_init(&revs->filter);
 }
 
 static void add_pending_commit_list(struct rev_info *revs,
diff --git a/revision.h b/revision.h
index afe1b77985f..93640aa30a0 100644
--- a/revision.h
+++ b/revision.h
@@ -357,7 +357,24 @@ struct rev_info {
  * called before release_revisions() the "struct rev_info" can be left
  * uninitialized.
  */
-#define REV_INFO_INIT { 0 }
+#define REV_INFO_INIT { \
+	.abbrev = DEFAULT_ABBREV, \
+	.simplify_history = 1, \
+	.pruning.flags.recursive = 1, \
+	.pruning.flags.quick = 1, \
+	.sort_order = REV_SORT_IN_GRAPH_ORDER, \
+	.dense = 1, \
+	.max_age = -1, \
+	.max_age_as_filter = -1, \
+	.min_age = -1, \
+	.skip_count = -1, \
+	.max_count = -1, \
+	.max_parents = -1, \
+	.expand_tabs_in_log = -1, \
+	.commit_format = CMIT_FMT_DEFAULT, \
+	.expand_tabs_in_log_default = 8, \
+	.filter = LIST_OBJECTS_FILTER_INIT, \
+}
 
 /**
  * Initialize a rev_info structure with default values. The third parameter may
-- 
2.38.0.1464.gea6794aacbc

